ProtocolHandler.pm

```diff
@@ -1,87 +1,148 @@
 package Plugins::RadioParadise::ProtocolHandler;
 
 use strict;
-# XXX - https
-use base qw(Slim::Player::Protocols::HTTP);
+use base qw(IO::Handle);
 
 use JSON::XS::VersionOneAndTwo;
 use Tie::Cache::LRU;
 use POSIX qw(ceil);
-
-use Slim::Networking::SimpleAsyncHTTP;
+use List::Util qw(min max);
+use AnyEvent::DNS;
 
 use Slim::Utils::Log;
 use Slim::Utils::Prefs;
 use Slim::Utils::Timers;
+use Slim::Utils::Errno;
+use Slim::Networking::Async::HTTP;
 
 use Plugins::RadioParadise::Stations;
 
-use constant BASE_URL => 'https://api.radioparadise.com/api/get_block?bitrate=%s&chan=%s&info=true%s';
-
-# skip very short segments, like eg. some announcements, they seem to cause timing or buffering issues
-use constant MIN_EVENT_LENGTH => 15;
+use constant BASE_URL 	=> 'https://api.radioparadise.com/api/get_block?bitrate=%s&chan=%s&info=true%s';
+use constant CHUNK_SIZE	=> 128*1024;
 
 my $prefs = preferences('plugin.radioparadise');
-tie my %blockData, 'Tie::Cache::LRU', 16;
-
-my %AAC_BITRATE = (
-	0 => 32,
-	1 => 64,
-	2 => 128,
-	3 => 320,
-);
-
 my $log = logger('plugin.radioparadise');
 
-# To support remote streaming (synced players, slimp3/SB1), we need to subclass Protocols::HTTP
+tie my %blockData, 'Tie::Cache::LRU', 16;
+
 sub new {
 	my $class  = shift;
 	my $args   = shift;
-
-	my $client = $args->{client};
-
-	my $song      = $args->{'song'};
-	my $streamUrl = $song->streamUrl() || return;
-
-	my ($quality, $format) = _getStreamParams( $args->{url} );
-
-	my $sock = $class->SUPER::new( {
-		url     => $streamUrl,
-		song    => $args->{'song'},
-		client  => $client,
-		bitrate => $quality < 4 ? $AAC_BITRATE{$quality} * 1024 : 850_000,
-	} ) || return;
-
-	${*$sock}{contentType} = 'audio/' . $format;
-
+	my $song = $args->{'song'};
+
+	my $streamUrl = $song->streamUrl || return;
+	my $sock = $class->SUPER::new;
+	my $session = Slim::Networking::Async::HTTP->new;
+
+	${*$sock}{'vars'}   = { 	       		# variables which hold state for this instance:
+			'url'		  => $streamUrl,	# Url to grab	
```

Comment:
Nitpicking alert: while you're still working on it, could you please make sure you don't mix spaces/tabs? Thanks! :-D

Reply:
yes, I know I'm bad at that :-( and I also think my main editor does not help in conditions I've not been able to understand

Reply:
What's your editor of choice?

Reply:
notepad++ :-)